scripts/generate_genotype_df/generate_genotype_df.py

```diff
@@ -29,15 +30,22 @@ def query():
     mt = mt.annotate_rows(freq=ht[mt.row_key].freq)
     mt = mt.filter_rows(mt.freq.AF[1] > 0.01)
     # select alleles and locus (contig and position must be selected separately),
-    # then convert to pandas
-    t = mt.rows()
+    t = mt.entries()
+    # get genotype codes (n_alt_alleles, as in PLINK files)
+    t = t.annotate(n_alt_alleles=t.GT.n_alt_alleles())
     t = t.key_by(contig=t.locus.contig, position=t.locus.position)
-    t = t.select(t.alleles)
-    pd = t.to_pandas(flatten=True)
-    # expand locus to two columns and rename 
+    t = t.select(t.alleles, t.s, t.n_alt_alleles)
```

Comment:
This will blow up the data frame _significantly_. This also won't scale well with larger number of samples. Maybe it's time to reconsider whether we shouldn't be implementing this all in Hail instead, even if that means we need to find an alternative to the Spearman correlation computation?

Reply:
Would it be easier to just export the PLINK files, as done originally?

Reply:
That won't really change the problem that you'd need to hold this complete table (or at least the section for one chromosome) in memory in a single node though, so fundamentally this seems to be very limited in terms of scalability.

I think if you really wanted to do this outside of Hail, you'd need to generate a DataFrame for much smaller regions and export them as files individually. Then you could run the association analysis on those individual chunks separately.

At that point you're kind of reimplementing some of the parallelization that Hail would take care of for you directly.

Reply:
I'm struggling to understand how this is so different from the way we ran the script previously, where we exported a parquet file for each chromosome?

Reply:
Didn't you previously only export site-level data, i.e. not per-sample genotypes?

I'm not saying that splitting into chunks is fundamentally different from splitting into chromosomes, but the amount of memory required per job is just way smaller if you can have much more fine-grained chunks.

For example, instead of exporting a file per chromosome here, could you instead launch 1000 jobs that each open the Hail MT, but process a chunk that's 1/1000 of the whole genome? Or maybe have a chunk per gene (with 1MB each side) or similar?

Reply:
(You could still do the processing using a Pandas DataFrame that way, but you'd skip exporting in a single-threaded way in this script completely.)

Reply:
Ok, I completely overlooked that this inflated the size so much. I'll look into chunking this